inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -571,22 +571,40 @@ def normalize_journal_titles(obj, eng):
     Returns:
        None
     """
-    publications = obj.data.get('publication_info')
-
-    if not publications:
-        return None
+    publications = obj.data.get('publication_info', [])
 
     for index, publication in enumerate(publications):
-        if 'journal_title' in publication:
-            normalized_title = normalize_journal_title(publication['journal_title'])
+        normalized_title, journal_record = normalize_journal_title_entry(publication)
+        if normalized_title:
             obj.data['publication_info'][index]['journal_title'] = normalized_title
-            ref_query = RecordMetadata.query.filter(
-                RecordMetadata.json['_collections'].op('?')('Journals')).filter(
-                cast(RecordMetadata.json['short_title'], String) == type_coerce(normalized_title, JSON))
-            result = db.session.execute(ref_query).fetchone()
-
-            if result:
-                obj.data['publication_info'][index]['journal_record'] = result.records_metadata_json['self']
+        if journal_record:
+            obj.data['publication_info'][index]['journal_record'] = journal_record
+
+    references = obj.data.get("references", [])
+    for index, reference in enumerate(references):
+        publication_info = get_value(reference, 'reference.publication_info')
+        if not publication_info:
+            continue
+        normalized_title, journal_record = normalize_journal_title_entry(publication_info)
+        if normalized_title:
+            obj.data['references'][index]['reference']['publication_info']['journal_title'] = normalized_title
```

Comment:
```suggestion
            reference['reference']['publication_info']['journal_title'] = normalized_title
```
and same thing on L592, so you don't need to `enumerate`.

Reply:
:+1: